addons/account/models/account_move.py

```diff
@@ -4703,277 +4703,336 @@ def _mail_track(self, tracked_fields, initial):
-                recon_debit_amount = remaining_debit_amount
-                recon_credit_amount = -remaining_credit_amount
+            # Compute the partial amount expressed in foreign currency.
+            if debit_rate:
+                partial_debit_amount_currency = debit_vals['currency'].round(debit_rate * min_recon_amount)
+                partial_debit_amount_currency = min(partial_debit_amount_currency, remaining_debit_amount_curr)
             else:
-                # Multiple involved foreign currencies. The reconciliation is done using the currency of the company.
-                recon_currency = company_currency
-                debit_rate = get_accounting_rate(debit_line)
-                credit_rate = get_accounting_rate(credit_line)
-                recon_debit_amount = remaining_debit_amount
-                recon_credit_amount = -remaining_credit_amount
-
-            # Check if there is something left to reconcile. Move to the next loop iteration if not.
-            skip_reconciliation = False
-            if recon_currency.is_zero(recon_debit_amount):
-                debit_line = None
-                skip_reconciliation = True
-            if recon_currency.is_zero(recon_credit_amount):
-                credit_line = None
-                skip_reconciliation = True
-            if skip_reconciliation:
-                continue
-
-            # ==== Match both lines together and compute amounts to reconcile ====
+                partial_debit_amount_currency = 0.0
+            if credit_rate:
+                partial_credit_amount_currency = credit_vals['currency'].round(credit_rate * min_recon_amount)
+                partial_credit_amount_currency = min(partial_credit_amount_currency, -remaining_credit_amount_curr)
+            else:
+                partial_credit_amount_currency = 0.0
 
-            # Determine which line is fully matched by the other.
-            compare_amounts = recon_currency.compare_amounts(recon_debit_amount, recon_credit_amount)
-            min_recon_amount = min(recon_debit_amount, recon_credit_amount)
-            debit_fully_matched = compare_amounts <= 0
-            credit_fully_matched = compare_amounts >= 0
+        else:
+            # recon_currency != company_currency
+            # Compute the partial amount expressed in company currency.
+            if debit_rate:
+                partial_debit_amount = company_currency.round(min_recon_amount / debit_rate)
+                partial_debit_amount = min(partial_debit_amount, remaining_debit_amount)
+            else:
+                partial_debit_amount = 0.0
+            if credit_rate:
+                partial_credit_amount = company_currency.round(min_recon_amount / credit_rate)
+                partial_credit_amount = min(partial_credit_amount, -remaining_credit_amount)
+            else:
+                partial_credit_amount = 0.0
+            partial_amount = min(partial_debit_amount, partial_credit_amount)
```

Comment:
Wouldn't it be better to simplify a bit this block like this?
It feels already a bit less complex. (+ the case looks quite similar)
```py
        # ==== Computation of partial amounts ====
        def compute_partial_amount(rate, currency, min_recon_amount, remaining_amount):
            if rate:
                partial_amount = min(currency.round(min_recon_amount * rate), remaining_amount)
            else:
                partial_amount = 0.0
            return partial_amount

        if recon_currency == company_currency:
            # Compute the partial amount expressed in foreign currency.
            partial_debit_amount_currency = compute_partial_amount(debit_rate, debit_vals['currency'], min_recon_amount, remaining_debit_amount_curr)
            partial_credit_amount_currency = compute_partial_amount(credit_rate, credit_vals['currency'], min_recon_amount, remaining_credit_amount_curr)
            # Compute the partial amount expressed in company currency.
            partial_amount = min(recon_debit_amount, recon_credit_amount)
        else:
            # recon_currency != company_currency
            # Compute the partial amount expressed in company currency.
            partial_debit_amount = compute_partial_amount(1 / debit_rate, debit_vals['currency'], min_recon_amount, remaining_debit_amount)
            partial_credit_amount = compute_partial_amount(1 / credit_rate, credit_vals['currency'], min_recon_amount, remaining_credit_amount)
            partial_amount = min(partial_debit_amount, partial_credit_amount)
```